exporter/clickhouseexporter/internal/metrics_model.go

```diff
@@ -0,0 +1,158 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/clickhouseexporter/internal"
+
+import (
+	"context"
+	"database/sql"
+	"encoding/hex"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+var supportMetricsType = [...]string{createGaugeTableSQL, createSumTableSQL, createHistogramTableSQL, createExpHistogramTableSQL, createSummaryTableSQL}
+
+type MetricsModel interface {
+	Add(metrics interface{}, name string, description string, unit string)
+	injectMetaData(metaData *MetricsMetaData)
+	insert(ctx context.Context, tx *sql.Tx, logger *zap.Logger) error
+}
+
+type MetricsMetaData struct {
+	ResAttr    map[string]string
+	ResURL     string
+	ScopeURL   string
+	ScopeInstr pcommon.InstrumentationScope
+}
+
+func CreateMetricsTable(tableName string, ttlDays uint, db *sql.DB) error {
+	var ttlExpr string
+	if ttlDays > 0 {
+		ttlExpr = fmt.Sprintf(`TTL toDateTime(TimeUnix) + toIntervalDay(%d)`, ttlDays)
+	}
+	for _, table := range supportMetricsType {
```

Comment:
isn't it wasteful to create all those tables per metric name, rather than based on the metric type used by the metric?

Reply:
it's does based on the metric type:
```
var supportMetricsType = [...]string{createGaugeTableSQL, createSumTableSQL, createHistogramTableSQL, createExpHistogramTableSQL, createSummaryTableSQL}
```

Reply:
it's based on the metric type:
```
var supportMetricsType = [...]string{createGaugeTableSQL, createSumTableSQL, createHistogramTableSQL, createExpHistogramTableSQL, createSummaryTableSQL}
```